### PR TITLE
Bug/workspace dataset user return lists

### LINKED ISSUE #4548: [BUG-python/deployment] argilla.Workspace.list() returning a generator and not a list
**Describe the bug**
argilla.Workspace.list(), User.list() and FeedbackDataset.list() returning a generator, nevertheless docstrings states that it shall return a list. We will make them return a list.

### PR DESCRIPTION
**Pull request by ignacioct**
Closes #4548 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Manually testing that Workspace, User and Feedback Dataset returns a list.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/users.py [hunk 1]
```diff
@@ -379,8 +379,6 @@ def list(cls) -> Iterator["User"]:
         """
         client = cls.__active_client()
         try:
-            users = users_api.list_users(client).parsed
-            for user in users:
-                yield cls.__new_instance(client, user)
+            return users_api.list_users(client).parsed
```
Comment (alvarobartt) [inline src/argilla/client/users.py]: Note that you're not returning a `User` here, but the `User` model from the SDK, which won't come with the methods and properties of the class, so this should be rewritten as

```suggestion
            return [cls.__new_instance(client, user) for user in users_api.list_users(client).parsed]
```

File: src/argilla/client/workspaces.py [hunk 2]
```diff
@@ -386,8 +386,7 @@ def list(cls) -> Iterator["Workspace"]:
         """
         client = cls.__active_client()
         try:
-            workspaces = workspaces_api_v1.list_workspaces_me(client).parsed
-            for ws in workspaces:
-                yield cls._new_instance(client, ws)
+            return workspaces_api_v1.list_workspaces_me(client).parsed
```
Comment (alvarobartt) [inline src/argilla/client/workspaces.py]: Same as above applies here!

**Review by alvarobartt — CHANGES_REQUESTED** [review]

File: src/argilla/client/users.py (hunk 1 above)
Comment (ignacioct) [reply src/argilla/client/users.py]: Thanks for noticing!

**Comment by ignacioct** [comment]
I think I fixed the problems @alvarobartt, these are the outputs:

`[Workspace(id=26c0b5d1-77cb-4494-bba8-c1406e4a27ab, name=argilla, inserted_at=2023-11-27 11:39:17.433020, updated_at=2023-11-27 11:39:17.433020), Workspace(id=ca00b60b-af26-462f-99f7-c04478fef100, name=awesome-argilla-datasets, inserted_at=2023-11-27 16:22:09.977232, updated_at=2023-11-27 16:22:09.977232)]`

`[User(id=895a7096-b7e9-4e4e-8729-a18a26792df5, username=owner, role=owner, api_key=..., first_name=owner, last_name=None, inserted_at=2023-11-27 11:35:54.681260, updated_at=2023-11-27 11:35:54.681260), ...]`

**Review by alvarobartt — APPROVED** [review]

**Pull request merged**